File: Listas dobles/Lista DOBLE.cs
using System;
using System.Collections.Generic;

class NodoDoble {
    public int valor;
    public NodoDoble siguiente;
    public NodoDoble anterior;
    
    public NodoDoble(int v) {
        valor = v;
        siguiente = null;
        anterior = null;
    }
}

class Program {
    static void MostrarLista(NodoDoble inicio, bool esAdelante = true) {
        NodoDoble actual = inicio;
        while (actual != null) {
            Console.Write(actual.valor + " ");
            actual = esAdelante ? actual.siguiente : actual.anterior;
        }
        Console.WriteLine();
    }

    static void Main() {
        int[] valores = {5, 10, 15, 20, 25};
        List<NodoDoble> nodos = new List<NodoDoble>();
        
        foreach (int v in valores) {
            nodos.Add(new NodoDoble(v));
        }
        
        for (int i = 0; i < nodos.Count; i++) {
            if (i > 0) {
                nodos[i].anterior = nodos[i-1];
            }
            if (i < nodos.Count - 1) {
                nodos[i].siguiente = nodos[i+1];
            }
        }
        
        Console.Write("Hacia adelante: ");
        MostrarLista(nodos[0]);
        
        Console.Write("Hacia atrás:    ");
        MostrarLista(nodos[nodos.Count-1], false);
    }
}

File: Listas dobles/Lista DOBLE.py
class NodoDoble:
    def __init__(self, valor):
        self.valor = valor
        self.siguiente = None
        self.anterior = None

def crear_lista(valores):
    nodos = [NodoDoble(v) for v in valores]
    for i in range(len(nodos)):
        if i > 0:
            nodos[i].anterior = nodos[i-1]
        if i < len(nodos) - 1:
            nodos[i].siguiente = nodos[i+1]
    return nodos

def mostrar_lista(inicio, direccion='adelante'):
    actual = inicio
    while actual:
        print(actual.valor, end=" ")
        actual = actual.siguiente if direccion == 'adelante' else actual.anterior
    print()

valores = [5, 10, 15, 20, 25, 30, 35, 40]
nodos = crear_lista(valores)
primero, *_, ultimo = nodos

print("Hacia adelante:", end=" ")
mostrar_lista(primero)

print("Hacia atrás:   ", end=" ")
mostrar_lista(ultimo, 'atras')
